chezmoi apply :: SIGSEGV :: 1.5.6 :: macOS

Running 'chezmoi apply' returns a 'SIGSEGV'. It looks like it has to do with inconsistent referencing 'once_' / 'run_once_' ? Removing the 'run_once_' prefixed script fixes the issue, however the use case requires the use of the 'run_once_' functionality provided by chezmoi. See below for output.

```
computer% pwd
/Users/username/.local/share/chezmoi
computer% ls -l                                                
total 32
-rw-r--r--  1 username  staff  36 Jun 16 18:00 run_050_giraffe
-rw-r--r--  1 username  staff  37 Jun 16 18:00 run_150_elephant
-rw-r--r--  1 username  staff  34 Jun 16 18:00 run_once_100_miauw.sh
computer% for file in $(ls); do echo "$file::"; cat $file; echo ''; done
run_050_giraffe::
#!/usr/bin/env bash

echo 'giraffe'

run_150_elephant::
#!/usr/bin/env bash

echo 'elephant'

run_once_100_miauw.sh::
#!/usr/bin/env bash

echo 'miauw'

computer% chezmoi apply
giraffe
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x137a696]

goroutine 1 [running]:
github.com/twpayne/chezmoi/lib/chezmoi.(*Script).Apply(0xc0000589c0, 0x17700a0, 0x1bd2800, 0x176c1c0, 0xc00007a440, 0xc000058a20, 0x0, 0x0)
	/home/travis/gopath/src/github.com/twpayne/chezmoi/lib/chezmoi/script.go:105 +0x936
github.com/twpayne/chezmoi/lib/chezmoi.(*TargetState).Apply(0xc0001425a0, 0x17700a0, 0x1bd2800, 0x176c1c0, 0xc00007a440, 0xc000058a20, 0x1bb6c00, 0x1e006d0)
	/home/travis/gopath/src/github.com/twpayne/chezmoi/lib/chezmoi/targetstate.go:156 +0xe8
github.com/twpayne/chezmoi/cmd.(*Config).applyArgs(0x1bad700, 0x17700a0, 0x1bd2800, 0x1bd2a58, 0x0, 0x0, 0x176c1c0, 0xc00007a440, 0xc000119d70, 0xc000119cf8)
	/home/travis/gopath/src/github.com/twpayne/chezmoi/cmd/config.go:147 +0x3b0
github.com/twpayne/chezmoi/cmd.(*Config).runApplyCmd(0x1bad700, 0x17700a0, 0x1bd2800, 0x1bd2a58, 0x0, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/twpayne/chezmoi/cmd/apply.go:23 +0x14f
github.com/twpayne/chezmoi/cmd.makeRunE.func1(0x1ba83e0, 0x1bd2a58, 0x0, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/twpayne/chezmoi/cmd/config.go:420 +0x5b
github.com/spf13/cobra.(*Command).execute(0x1ba83e0, 0x1bd2a58, 0x0, 0x0, 0x1ba83e0, 0x1bd2a58)
	/home/travis/gopath/pkg/mod/github.com/0robustus1/cobra@v0.0.4-0.20190522074606-64400adf086c/command.go:762 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0x1ba8fc0, 0x0, 0x0, 0x15360d3)
	/home/travis/gopath/pkg/mod/github.com/0robustus1/cobra@v0.0.4-0.20190522074606-64400adf086c/command.go:850 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/travis/gopath/pkg/mod/github.com/0robustus1/cobra@v0.0.4-0.20190522074606-64400adf086c/command.go:800
github.com/twpayne/chezmoi/cmd.Execute()
	/home/travis/gopath/src/github.com/twpayne/chezmoi/cmd/root.go:112 +0x2e
main.main()
	/home/travis/gopath/src/github.com/twpayne/chezmoi/main.go:6 +0x20
computer% 
```

--- a/cmd/root.go
+++ b/cmd/root.go
@@ -90,16 +90,20 @@ func init() {
 	_ = viper.BindPFlag("color", persistentFlags.Lookup("color"))
 
 	cobra.OnInitialize(func() {
-		if _, err := os.Stat(config.configFile); os.IsNotExist(err) {
-			return
-		}
-		viper.SetConfigFile(config.configFile)
-		config.err = viper.ReadInConfig()
-		if config.err == nil {
-			config.err = viper.Unmarshal(&config)
-		}
-		if config.err != nil {
-			config.warn(fmt.Sprintf("%s: %v", config.configFile, config.err))
+		_, err := os.Stat(config.configFile)
+		switch {
+		case err == nil:
+			viper.SetConfigFile(config.configFile)
+			config.err = viper.ReadInConfig()
+			if config.err == nil {
+				config.err = viper.Unmarshal(&config)
+			}
+			if config.err != nil {
+				config.warn(fmt.Sprintf("%s: %v", config.configFile, config.err))
+			}
+		case os.IsNotExist(err):
+		default:
+			printErrorAndExit(err)
 		}
 		persistentStateFile := getPersistentStateFile(config.bds, config.configFile)
 		persistentState, err := chezmoi.NewBoltPersistentState(vfs.OSFS, persistentStateFile)

--- a/docs/HOWTO.md
+++ b/docs/HOWTO.md
@@ -454,7 +454,9 @@ Scripts break chezmoi's declarative approach, and as such should be used
 sparingly. Any script should be idempotent, even `run_once_` scripts.
 
 Scripts must be created manually in the source directory, typically by running
-`chezmoi cd` and then creating a file with a `run_` prefix.
+`chezmoi cd` and then creating a file with a `run_` prefix. Scripts are executed
+directly using `exec` and must include a shebang line or be executable binaries.
+There is no need to set the executable bit on the script.
 
 Scripts with the suffix `.tmpl` are treated as templates, with the usual
 template variables available. If, after executing the template, the result is

